class Foo(var bar : Int, var barr : Int, var barrr : Int) {
    {
      bar = 1
      barr = 1
      barrr = 1
      1 : Int
      this : Foo
    }

    <!SECONDARY_CONSTRUCTORS_ARE_NOT_SUPPORTED!>this(var bar : Int) : this(1, 1, 1) {
      bar = <!UNUSED_VALUE!>1<!>
      this.bar
      1 : Int
      val <!UNUSED_VARIABLE!>a<!> : Int =1
      this : Foo
    }<!>
  }